internal/service/ecs/service.go

```diff
@@ -2281,195 +2291,347 @@ func expandPlacementStrategy(s []any) ([]awstypes.PlacementStrategy, error) {
-		config.TagSpecifications = expandTagSpecifications(ctx, v)
+	if v, ok := tfMap[names.AttrVolumeType].(string); ok && v != "" {
+		apiObject.VolumeType = aws.String(v)
 	}
 
-	return config
+	return apiObject
 }
 
-func expandTagSpecifications(ctx context.Context, ts []any) []awstypes.EBSTagSpecification {
-	if len(ts) == 0 {
+func expandEBSTagSpecifications(ctx context.Context, tfList []any) []awstypes.EBSTagSpecification {
+	if len(tfList) == 0 {
 		return []awstypes.EBSTagSpecification{}
 	}
 
-	var s []awstypes.EBSTagSpecification
-	for _, item := range ts {
-		raw, ok := item.(map[string]any)
+	var apiObjects []awstypes.EBSTagSpecification
+
+	for _, tfMapRaw := range tfList {
+		tfMap, ok := tfMapRaw.(map[string]any)
 		if !ok {
 			continue
 		}
 
-		var config awstypes.EBSTagSpecification
-		if v, ok := raw[names.AttrResourceType].(string); ok && v != "" {
-			config.ResourceType = awstypes.EBSResourceType(v)
+		var apiObject awstypes.EBSTagSpecification
+
+		if v, ok := tfMap[names.AttrPropagateTags].(string); ok && v != "" {
+			apiObject.PropagateTags = awstypes.PropagateTags(v)
 		}
-		if v, ok := raw[names.AttrPropagateTags].(string); ok && v != "" {
-			config.PropagateTags = awstypes.PropagateTags(v)
+		if v, ok := tfMap[names.AttrResourceType].(string); ok && v != "" {
+			apiObject.ResourceType = awstypes.EBSResourceType(v)
 		}
-		if v, ok := raw[names.AttrTags].(map[string]any); ok && len(v) > 0 {
+		if v, ok := tfMap[names.AttrTags].(map[string]any); ok && len(v) > 0 {
 			if v := tftags.New(ctx, v).IgnoreAWS(); len(v) > 0 {
-				config.Tags = svcTags(v)
+				apiObject.Tags = svcTags(v)
 			}
 		}
-		s = append(s, config)
+
+		apiObjects = append(apiObjects, apiObject)
 	}
 
-	return s
+	return apiObjects
+}
+
+func flattenServiceVolumeConfigurations(ctx context.Context, apiObjects []awstypes.ServiceVolumeConfiguration) []any {
+	if len(apiObjects) == 0 {
+		return nil
+	}
+
+	tfList := make([]any, 0, len(apiObjects))
+
+	for _, apiObject := range apiObjects {
+		tfMap := map[string]any{
+			names.AttrName: aws.ToString(apiObject.Name),
+		}
+
+		if v := apiObject.ManagedEBSVolume; v != nil {
+			tfMap["managed_ebs_volume"] = []any{flattenServiceManagedEBSVolumeConfiguration(ctx, v)}
+		}
+
+		tfList = append(tfList, tfMap)
+	}
+
+	return tfList
+}
+
+func flattenServiceManagedEBSVolumeConfiguration(ctx context.Context, apiObject *awstypes.ServiceManagedEBSVolumeConfiguration) map[string]any {
+	tfMap := map[string]any{
+		names.AttrRoleARN: aws.ToString(apiObject.RoleArn),
+	}
+
+	if v := apiObject.Encrypted; v != nil {
+		tfMap[names.AttrEncrypted] = aws.ToBool(v)
+	}
+	if v := apiObject.FilesystemType; v != "" {
+		tfMap["file_system_type"] = v
+	}
+	if v := apiObject.Iops; v != nil {
+		tfMap[names.AttrIOPS] = aws.ToInt32(v)
+	}
+	if v := apiObject.KmsKeyId; v != nil {
+		tfMap[names.AttrKMSKeyID] = aws.ToString(v)
+	}
+	if v := apiObject.RoleArn; v != nil {
+		tfMap[names.AttrRoleARN] = aws.ToString(v)
+	}
```

Comment:
@ewbankkit this is already assigned when initializing `tfMap` as it's a required field in [`ServiceManagedEBSVolumeConfiguration`](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ServiceManagedEBSVolumeConfiguration.html#API_ServiceManagedEBSVolumeConfiguration_Contents); we should only keep one of these assignments

Reply:
@ewbankkit for clarification, right now the `role_arn` attribute is being redundantly assigned twice:
```go
	tfMap := map[string]any{
		names.AttrRoleARN: aws.ToString(apiObject.RoleArn),
	}
...
	if v := apiObject.RoleArn; v != nil {
		tfMap[names.AttrRoleARN] = aws.ToString(v)
	}
```